Af 40 test data

BartekSzymik opened the pull request:
# Description

Preparing test data for decks, not yet flashcard, we need to write model for them.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
It has been tested manually and pytest

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The related memme has been added

![alt memme](https://media.giphy.com/media/uTz3P3wsV52l8wHrDU/giphy.gif)

SlawCzech committed:
fetch player profile data

martynawitkowska committed:
AQ-80 Profile data models (#59)

SlawCzech committed:
AQ-83 player api view (#56)

BartekSzymik committed:
generating test data for decks batch

aluczakzero reviewed (approved):
Seems to work fine, lacks only factory for Flashcard but Flashcard model is not done yet, so it will have to be added later, for now, Good job!

MWasile reviewed (changes requested):
use pytest for testing

maciejpiszczek commented on flashcards/backend/decks/tests/test_decks.py:
I would suggest using pytest for tests.

BartekSzymik replied:
now, should be good, i used command 'docker compose exec backend pytest -k decks'

maciejpiszczek reviewed (changes requested):
Minor (not so significant) changes to do but everything works fine. Good job! :)

ToJestKrzysio committed:
Aq 103 setup monorepo (#58)

* AQ-103 Update gitignore

* AQ-103 Trying random stuff

* AQ-103 Repo works without docker and without tailwind

* AQ-103 Fixed tailwind

* AQ-103 Removed test packages

* A lot of magic

* AQ-103 Cleanup

* AQ-103 WORKSSSSS!!!!

* AQ-103 Undo a f-up

* AQ-103 Fixed problem with mobile

* AQ-103 Fixed all problems.

* AQ-103 Fixed small mistake

* AQ-103 Update tailwind and postcss
Changed tailwind.config.cjs to tailwind.config.js
Changed postcss.config.cjs to postcss.config.js

* AQ-103 Fix dev

* AQ-103 Fix storybook

* AQ-103 Small tweak

martynawitkowska committed:
AQ-79 badge endpoint (#65)

annagorkiewicz committed:
update django to version 4.2 LTS with psycopg3 support, psycopg update to version 3.1.8 (#63)

BartekSzymik commented on flashcards/backend/decks/factories.py:
```diff
@@ -0,0 +1,76 @@
+from factory import Faker, Sequence, SubFactory
+
+from backend.decks.models import Category, Flashcard, Deck, DifficultyLevel, Tag
+from backend.users.factories import UserFactory
```
its done now, i had to set sources root

BartekSzymik commented:
@MWasile, i have fixed test

MWasile commented:
still not in pytest

BartekSzymik commented:
@MWasile, what do you mean still not in pytest? You mean there is not enough test written? When I make pytest all test pass, when I make pytest -k decks there is a mistake of coverage? Can you suggest what to do?

martynawitkowska commented on flashcards/backend/decks/tests/test_decks.py:
```diff
@@ -0,0 +1,46 @@
+from decks.factories import CategoryFactory, DeckFactory, FlashcardFactory, TagFactory, DifficultyLevelFactory
+
+
+class TestFactories(TestCase):
+    def test_category_factory(self):
+        """Test CategoryFactory."""
+        category = CategoryFactory()
+        self.assertIsNotNone(category)
+        self.assertIsNotNone(category.name)
+        self.assertIsNotNone(category.description)
+
```
@BartekSzymik what @MWasile is saying is that you use UnitTest for testing and our standard is pytest. Those tests are only passing because pytest can run also UnitTest cases. You need rewrite this file using pytest package and not UniTest. I will only say that as we can build class-based tests with pytest, it is strongly recommended to use functions for testing ;)

BartekSzymik replied:
Thank you Martyna

BartekSzymik commented:
@MWasile now should be good, what do you think?

martynawitkowska commented:
Something went wrong, because it looks like you have worked on files from `bank` directory. As I remember you are working only in backend directory. 

<img width="287" alt="Screenshot 2023-05-17 at 15 23 38" src="https://github.com/AzenoIT/azeno/assets/92261595/516ce567-12da-44cd-96f2-b4502b343a0b">
